### PR TITLE
Gateway: Improve documentation for query parameters

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

Gateway: Improve documentation for query parameters

## How is this patch tested?

- [X] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [X] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/gateway/index.rst [hunk 1]
```diff
@@ -518,7 +518,7 @@ OpenAI
 ++++++
 
 +-------------------------+----------+-------------------------------+-------------------------------------------------------------+
-| Parameter               | Required | Default                       | Description                                                 |
+| Configuration Parameter | Required | Default                       | Description                                                 |
```
Comment (dbczumar) [inline docs/source/gateway/index.rst:521]: Distinguish configuration parameters from query parameters

File: docs/source/gateway/index.rst [hunk 2]
```diff
@@ -215,6 +215,13 @@ At this point, you may use the :ref:`gateway_fluent_api` to query both routes wi
 
 If you no longer need a route, you can delete it from the configuration YAML and save your changes. The AI Gateway will automatically remove the route.
 
+Step 10: Use AI Gateway routes for model development
+----------------------------------------------------
+
+Now that you have created several AI Gateway routes, you can create MLflow Models that query these
+routes to build application-specific logic using techniques like prompt engineering. For more
+information, see :ref:`AI Gateway and MLflow Models <gateway_mlflow_models>`.
```
Comment (dbczumar) [inline docs/source/gateway/index.rst:223]: This is super important. We should highlight it in the quickstart

File: docs/source/gateway/index.rst [hunk 3]
```diff
@@ -623,18 +630,91 @@ Once the MLflow AI Gateway server has been configured and started, it is ready t
...
+| **temperature**               | float          | No       | 0.0           | The sampling temperature to use, between 0 and 1.     |
+|                               |                |          |               | Higher values will make the output more random, and   |
+|                               |                |          |               | lower values will make the output more deterministic. |
++-------------------------------+----------------+----------+---------------+-------------------------------------------------------+
+| **max_tokens**                | integer        | No       | infinity      | The maximum completion length, between 1 and infinity |
+|                               |                |          |               | (unlimited).                                          |
++-------------------------------+----------------+----------+---------------+-------------------------------------------------------+
+| **stop**                      | array[string]  | No       | []            | Sequences where the model should stop generating      |
```
Comment (harupy) [inline docs/source/gateway/index.rst:685]: The default stop value is None. An empty array is invalid for OpenAI and results in a bad request.

File: docs/source/gateway/index.rst [hunk 4]
```diff
@@ -623,18 +630,91 @@ Once the MLflow AI Gateway server has been configured and started, it is ready t
...
+| **candidate_count**           | integer        | No       | 1             | The number of chat completions to generate for the    |
+|                               |                |          |               | specified prompt, between 1 and 5.                    |
++-------------------------------+----------------+----------+---------------+-------------------------------------------------------+
+| **temperature**               | float          | No       | 0.0           | The sampling temperature to use, between 0 and 1.     |
+|                               |                |          |               | Higher values will make the output more random, and   |
+|                               |                |          |               | lower values will make the output more deterministic. |
++-------------------------------+----------------+----------+---------------+-------------------------------------------------------+
+| **max_tokens**                | integer        | No       | infinity      | The maximum completion length, between 1 and infinity |
```
Comment (harupy) [inline docs/source/gateway/index.rst:682]:
```suggestion
| **max_tokens**                | integer        | No       | None      | The maximum completion length, between 1 and infinity |
```

The default `max_tokens` value is None.

File: docs/source/gateway/index.rst (hunk 3 above)
Comment (harupy) [reply docs/source/gateway/index.rst:685]:
```suggestion
| **stop**                      | array[string]  | No       | None            | Sequences where the model should stop generating      |
```

File: docs/source/gateway/index.rst [hunk 5]
```diff
@@ -623,18 +630,91 @@ Once the MLflow AI Gateway server has been configured and started, it is ready t
...
+| Query Parameter               | Type           | Required | Default       | Description                                           |
++===============================+================+==========+===============+=======================================================+
+| **messages**                  | array[message] | Yes      | N/A           | A list of messages in a conversation from which to    |
+|                               |                |          |               | a new message (chat completion). For information      |
+|                               |                |          |               | about the message structure, see                      |
+|                               |                |          |               | :ref:`chat_message_structure`.                        |
++-------------------------------+----------------+----------+---------------+-------------------------------------------------------+
+| **candidate_count**           | integer        | No       | 1             | The number of chat completions to generate for the    |
```
Comment (harupy) [inline docs/source/gateway/index.rst:675]:
```suggestion
| **candidate_count**           | integer        | No       | None             | The number of chat completions to generate for the    |
```

File: docs/source/gateway/index.rst [hunk 6]
```diff
@@ -623,18 +630,91 @@ Once the MLflow AI Gateway server has been configured and started, it is ready t
...
-
-- ``temperature``: This parameter controls the randomness of predictions by scaling the logits before applying softmax. A value closer to 0.0 makes the output more deterministic, while a value closer to 1.0 makes it more diverse. Default is 0.0.
-
-- ``max_tokens``: This parameter limits the length of the generated output by specifying a maximum token count. The range is from 1 to infinity, and by default, there is no limit (infinity). Some providers have a maximum value associated with this parameter that the AI Gateway will enforce to prevent a provider-generated exception.
+The MLflow AI Gateway defines standard parameters for chat, completions, and embeddings that can be
+used when querying any route regardless of its provider. Each parameter has a standard range and
+default value. When querying a route with a particular provider, the MLflow AI Gateway automatically
+scales parameter values according to the provider's value ranges for that parameter.
+
+Completions
```
Comment (harupy) [inline docs/source/gateway/index.rst:638]: We also need fix the default values in completions.

File: docs/source/gateway/index.rst [hunk 7]
```diff
@@ -654,9 +734,11 @@ Below is an example of submitting a query request to an MLflow AI Gateway route
...
 
     data = {
-        "prompt": "What would happen if an asteroid the size of "
-        "a basketball encountered the Earth traveling at 0.5c? "
-        "Please provide your answer in .rst format for the purposes of documentation.",
+        "prompt": (
+            "What would happen if an asteroid the size of "
+            "a basketball encountered the Earth traveling at 0.5c? "
+            "Please provide your answer in .rst format for the purposes of documentation."
+        ),
```
Comment (harupy) [inline docs/source/gateway/index.rst:741]: to improve readability

File: docs/source/gateway/index.rst [hunk 8]
```diff
@@ -623,18 +630,91 @@ Once the MLflow AI Gateway server has been configured and started, it is ready t
...
-If a given provider does not provide support for a parameter, a clear message will be returned when queried that explains the restrictions for the given provider and route type.
-
-- ``temperature``: This parameter controls the randomness of predictions by scaling the logits before applying softmax. A value closer to 0.0 makes the output more deterministic, while a value closer to 1.0 makes it more diverse. Default is 0.0.
-
-- ``max_tokens``: This parameter limits the length of the generated output by specifying a maximum token count. The range is from 1 to infinity, and by default, there is no limit (infinity). Some providers have a maximum value associated with this parameter that the AI Gateway will enforce to prevent a provider-generated exception.
+The MLflow AI Gateway defines standard parameters for chat, completions, and embeddings that can be
+used when querying any route regardless of its provider. Each parameter has a standard range and
+default value. When querying a route with a particular provider, the MLflow AI Gateway automatically
+scales parameter values according to the provider's value ranges for that parameter.
```
Comment (harupy) [inline docs/source/gateway/index.rst:636]: An example might clarify what this means. For example, we could say "For example, the temperature will be multiplied by 2 for OpenAI".

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**